committeeoversightapp/management/commands/import_data.py

```diff
@@ -145,18 +151,30 @@ def add_senate_hearings(self):
                 participating_committees = [committee for committee in committees if committee]
 
                 if name:
-                    self.matched = self.does_hearing_exist(name, start_date, classification)
-
-                    if (not self.matched) and hearing_number_raw:
-                        self.match_by_hearing_number(hearing_number_raw, name, category, source)
+                    print("\nSenate row " + str(self.row_count) + ": " + name)
+                    exists = self.does_hearing_exist(name, start_date, classification, category)
 
-                    if not self.matched:
-                        self.match_by_date_and_participants(name, participating_committees, start_date, category, source, classification)
+                    if exists:
+                        print("Already exists!")
+                        self.noop_count += 1
 
-                    if not self.matched:
-                        self.create_hearing(name, start_date, participating_committees, classification, category, source)
+                    else:
+                        events = self.match_by_hearing_number(hearing_number_raw, name, category, source)
+                        if events is not None:
+                            # hearing numbers can be non-unique if a hearing has multiple sessions
+                            # these are recorded in the Lugar data as separate events
+                            for event in events:
+                                self.update_hearing(event, category, source)
+                                break
+
+                        event = self.match_by_date_and_participants(name, participating_committees, start_date, category, source, classification)
```

Comment:
should we really try this match path regardless of the return value of match_by_hearing_number?

Reply:
true, we should stop once a hearing number is found! I'll update.

Reply:
_Updated_